src/Symfony/Bridge/Monolog/Handler/MailerHandler.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Handler;
+
+use Monolog\Formatter\FormatterInterface;
+use Monolog\Formatter\HtmlFormatter;
+use Monolog\Formatter\LineFormatter;
+use Monolog\Handler\MailHandler;
+use Monolog\Logger;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+
+/**
+ * @author Alexander Borisov <boshurik@gmail.com>
+ */
+class MailerHandler extends MailHandler
+{
+    private $mailer;
+
+    private $messageTemplate;
+
+    /**
+     * @param callable|Email $message
+     */
+    public function __construct(MailerInterface $mailer, $message, int $level = Logger::DEBUG, bool $bubble = true)
+    {
+        parent::__construct($level, $bubble);
+
+        $this->mailer = $mailer;
+        $this->messageTemplate = $message;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function send($content, array $records)
+    {
+        $this->mailer->send($this->buildMessage($content, $records));
+    }
+
+    /**
+     * Gets the formatter for the Message subject.
+     *
+     * @param string $format The format of the subject
+     */
+    protected function getSubjectFormatter(string $format): FormatterInterface
+    {
+        return new LineFormatter($format);
+    }
+
+    /**
+     * Creates instance of Message to be sent.
+     *
+     * @param string $content formatted email body to be sent
+     * @param array  $records Log records that formed the content
+     */
+    protected function buildMessage(string $content, array $records): Email
+    {
+        $message = null;
+        if ($this->messageTemplate instanceof Email) {
+            $message = clone $this->messageTemplate;
+        } elseif (\is_callable($this->messageTemplate)) {
+            $message = \call_user_func($this->messageTemplate, $content, $records);
+        }
+
+        if (!$message instanceof Email) {
+            throw new \InvalidArgumentException('Could not resolve message as instance of Email or a callable returning it');
+        }
```

Comment:
This probably makes more sense up inside the `elseif`. And then, I would add a final `else` to throw an exception. Basically:

```
        if ($this->messageTemplate instanceof Email) {
            $message = clone $this->messageTemplate;
        } elseif (\is_callable($this->messageTemplate)) {
            $message = \call_user_func($this->messageTemplate, $content, $records);
            if (!$message instanceof Email) {
                throw new \InvalidArgumentException('Callable must reutrn an Email instance');
            }
        } else {
            throw new \Exception('Invalid message argument');
        }
```